Assignment5 - Alexa Darrington and Johanne McClenahan

JohanneJayde opened the pull request:
Alexa's Github - @alyxadarr
Johanne's Github - @JohanneJayde

Website URL: https://aestheticwordle.com/

| Acount Type | Email                   | Password    |
| ----------- | ----------------------- | ----------- |
| Admin       | Admin@Intellitect.com   | P@ssw0rd123 |
| Awesome     | Awesome@Intellitect.com | P@ssw0rd123 |

Awesome should be able to update common word flag, delete word, and add word
Admin should only be able to update the common word flag

alyxadarr committed:
Updated main_aestheticwordle.yml

JohanneJayde committed:
updated table format

JohanneJayde committed:
updated api to use PlayerDto

JohanneJayde committed:
added in rank in leaderboard

JohanneJayde committed:
SaveScore endpoint functions now

JohanneJayde committed:
Update main_aestheticwordle.yml

JohanneJayde committed:
Update README.md

alyxadarr committed:
Added Average Seconds to Player.cs, Score.cs, PlayerDto.cs

JohanneJayde committed:
added mor styling to leaderobard

JohanneJayde committed:
fixed validWords pagination issue

alyxadarr committed:
More Seconds implementation

alyxadarr committed:
Tried adding AverageSeconds to leaderboard.vue

JohanneJayde committed:
fixed migrations

JohanneJayde committed:
timer added to site

JohanneJayde committed:
updated request body in saveScore to match Score model

JohanneJayde committed:
Revert "fixed migrations"

This reverts commit 9edaccd3facd39485c4e8c51bf1ace3e8315641d.

JohanneJayde committed:
added in migration for AverageSeconds

JohanneJayde committed:
stopwtch to stopwatch

JohanneJayde committed:
Sort leaderboards on AverageSeconds as well

JohanneJayde committed:
leaderboard entries are now colored based on theme background

alyxadarr committed:
Added comment to tets

alyxadarr committed:
Removed comment

alyxadarr committed:
Added better quality svg for logo. cleaned up extra files in public

JohanneJayde committed:
added icons to player name and timer

alyxadarr committed:
Added logo to about page and more styling

JohanneJayde committed:
Score is no longer recorded when you lose the game

JohanneJayde committed:
updating UI styling

JohanneJayde committed:
made player name larger in leaderboard

JohanneJayde committed:
more UI updates

JohanneJayde committed:
converted layout to grid based for index

JohanneJayde committed:
WordList component now only has dialog and not button

JohanneJayde committed:
simplified NameDialog logic

JohanneJayde committed:
Added ability to restart the game

JohanneJayde committed:
resetting stopwatch now stops to clear interval then sets time back to 0

JohanneJayde committed:
leaderboard title matches background color now

JohanneJayde committed:
updated method names for saving player data

JohanneJayde committed:
initial commit

JohanneJayde committed:
ci: add Azure Static Web Apps workflow file
on-behalf-of: @Azure opensource@microsoft.com

JohanneJayde committed:
updated yml files

JohanneJayde committed:
deploy api when assignment4 has PR

JohanneJayde committed:
added new azure site URL

JohanneJayde committed:
minor formatting update

JohanneJayde committed:
book for valid words sheet in index

JohanneJayde committed:
added instructions and daily wordle pages

JohanneJayde committed:
fixed v-img tag

JohanneJayde committed:
Merge remote-tracking branch 'upstream/main' into Assignment4

JohanneJayde committed:
converted game to reactive instead of ref again

JohanneJayde committed:
loading status works on index page

JohanneJayde committed:
Daily Wordle page now renders

JohanneJayde committed:
commnted out tests for now

JohanneJayde committed:
commented out wordlist tests

JohanneJayde committed:
fixed game and wordlist tests

JohanneJayde committed:
added latestWordle.vue

JohanneJayde committed:
Merge branch 'IntelliTect-Samples:main' into main

JohanneJayde committed:
Update README.md

JohanneJayde committed:
updated npm packages

JohanneJayde committed:
WodOfTheDayId is no longer required for posting game

JohanneJayde committed:
updating migrations

JohanneJayde committed:
aded getting game stats for a given word

JohanneJayde committed:
updated logic for posting games

JohanneJayde committed:
added UI for getting latest ten wordles and also updated controller and services

JohanneJayde committed:
now using tabs for displaying daily wordle stats

JohanneJayde committed:
WordStasDto now returns list of GameDtos

JohanneJayde committed:
main wordle view is now responsive

JohanneJayde committed:
app bar buttons are not nav icons

JohanneJayde committed:
leaderboard and about are responsive snow

JohanneJayde committed:
remove compact desnity from v-table

alyxadarr committed:
Changed vuetify.ts for word choices to be consistent in the themes.

alyxadarr committed:
Added in hover and truncate for username

alyxadarr committed:
Fixed truncate value to 8

alyxadarr committed:
Centered color codes in instructions page

alyxadarr committed:
Added styling to instructions.vue and added color codes

JohanneJayde committed:
Use v-tooltip instead of v expansion

JohanneJayde committed:
remove v-hover and extra v-bind

alyxadarr committed:
Restyled instructions page and added second page

alyxadarr committed:
Logo now dynamically changes with the theme

alyxadarr committed:
Fixed logo in about.vue to dynamically change too

alyxadarr committed:
Merge branch 'IntelliTect-Samples:main' into main

alyxadarr committed:
Added migration for game seconds for post

alyxadarr committed:
Migration for seconds

alyxadarr committed:
Added landing page for game mode

JohanneJayde committed:
Merge remote-tracking branch 'upstream/main'

JohanneJayde committed:
Merge branch 'main' into Assignment4

JohanneJayde committed:
fix merge issues

JohanneJayde committed:
fixed tests

JohanneJayde committed:
added WordOfTheDayStsats endpoint and updating UI

JohanneJayde committed:
added in WordOfTheDayStatsCard.vue

JohanneJayde committed:
fix backend to post and track seconds for games

JohanneJayde committed:
WOTDStatsCard retrieves stats correctly now

JohanneJayde committed:
updated LatestWordles to use new controller and service

JohanneJayde committed:
LatestWordle now uses gird based responsiveness

JohanneJayde committed:
minor format updates

alyxadarr committed:
changed spelling of random

alyxadarr committed:
Added gradient animation and restyled landing page

alyxadarr committed:
changed spelling on percentage

JohanneJayde committed:
Radnom Worlde and Daily Worlde now function

JohanneJayde committed:
dynamic routing words now

JohanneJayde committed:
Can now play from  past ten wordles

JohanneJayde committed:
added has played flag in latest wordle

JohanneJayde committed:
Side drawer links work but are ugly

JohanneJayde committed:
removed layout file for now

JohanneJayde committed:
remove extra console log out

JohanneJayde committed:
remove broken tests

JohanneJayde committed:
remove invalid param from startNewGame

JohanneJayde committed:
made test async

JohanneJayde committed:
Player name and seconds are name saved when posting results

JohanneJayde committed:
tracking if user has completed a given word of the day now

JohanneJayde committed:
temp fixed to issue with date being offset by 1

JohanneJayde committed:
format URLs

JohanneJayde committed:
add tracking name in game table (migration)

JohanneJayde committed:
fixed ugly route in app.vue

JohanneJayde committed:
Rename LatestWordles.vue to latestwordles.vue

fixed caps for routing

JohanneJayde committed:
update date paramter to be uniform

JohanneJayde committed:
update win percentage

JohanneJayde committed:
make word stats smaller

JohanneJayde committed:
fix win percentage

JohanneJayde committed:
updating average guess and win percentage size

alyxadarr committed:
Added gradient to all pages

alyxadarr committed:
fixing check out leaderboard button

alyxadarr committed:
fixing check out leaderboard button

alyxadarr committed:
Added responsiveness

alyxadarr committed:
Changed button for leaderboard

alyxadarr committed:
adding temp background for latest wordles

alyxadarr committed:
removed background from app.vue

alyxadarr committed:
removed style in app.vue

meghanmae committed:
Update Assignment5.md

alyxadarr committed:
Fixed animation

alyxadarr committed:
Johanne fixed animation layout

meghanmae committed:
Create signin dialog

JohanneJayde committed:
pages are now lowercase

JohanneJayde committed:
update route in WordStatsCard

JohanneJayde committed:
fix routes casing in side drawer

JohanneJayde committed:
more route fixes

JohanneJayde committed:
re fixed routes so they are all capital case

meghanmae committed:
Finalize login dialog and add a token service

JohanneJayde committed:
temp fix to update random route in side drawer

meghanmae committed:
Make authenticated get endpoint

alyxadarr committed:
update gradient

alyxadarr committed:
WIP on Assignment4

alyxadarr committed:
Merge branch 'Assignment4' of https://github.com/JohanneJayde/EWU-CSCD379-2024-Spring into Assignment4

alyxadarr committed:
Fixed font

alyxadarr committed:
Reduced font size for daily wordle

JohanneJayde committed:
updated Wordle games to use Game.vue intsead of copy pasting components

JohanneJayde committed:
fix names for daily and raomd wordle in side drawer navigation

JohanneJayde committed:
remove extra div in latest wordle page

alyxadarr committed:
fixed responsiveness with misplaced block

JohanneJayde committed:
created dateUtils for formatting dates within app

GrantErickson committed:
Update Lectures.md

GrantErickson committed:
Update Lectures.md

JohanneJayde committed:
re add in deleted scripts from WordList.vue

JohanneJayde committed:
undo breaking choosing username

JohanneJayde committed:
updated timer for mobile display to so it doesn't take up space

alyxadarr committed:
fix wordle stats card link

GrantErickson committed:
Policies

alyxadarr committed:
Merge remote-tracking branch 'upstream/main' into Assignment5

JohanneJayde committed:
ci: add Azure Static Web Apps workflow file
on-behalf-of: @Azure opensource@microsoft.com

JohanneJayde committed:
update workflow for Assignment5

JohanneJayde committed:
update elevation on wordle stats card

JohanneJayde committed:
update README.md for A5

JohanneJayde committed:
Added word Editor page and all pages are now capital case

JohanneJayde committed:
Delted extra space in app menu item

JohanneJayde committed:
rename instruction to instructions

JohanneJayde committed:
fixed references to pages still being lowercase

JohanneJayde committed:
Using tabs instead of having two insutrctions pages

JohanneJayde committed:
fix rename for LatestWordles

JohanneJayde committed:
Added WordsList endpoint for getting words via the server

JohanneJayde committed:
added isCommonWord to Word (Migration)

JohanneJayde committed:
wordsList is now displayed in WordEditor

JohanneJayde committed:
Added WordEditorRow for showing words

JohanneJayde committed:
update to add in delete button

JohanneJayde committed:
Fix deploy step not working by updating to assignment5 branch

JohanneJayde committed:
fix spelling mistake for GetRandomWord

JohanneJayde committed:
Update README.md

JohanneJayde committed:
Update landing page layout

JohanneJayde committed:
added new theme picker

JohanneJayde committed:
shorten theme names

JohanneJayde committed:
updating Ruby theme and fixed issue with thmes not updating on reload

JohanneJayde committed:
added back in default theme

JohanneJayde committed:
Theme Picker handles if dark mode is on

JohanneJayde committed:
change theme picker orientation if on mobile

JohanneJayde committed:
added nifty descriptions for each game mode

JohanneJayde committed:
using data table for now for word editor

JohanneJayde committed:
sitched to data-table-server

JohanneJayde committed:
added temp favicon

JohanneJayde committed:
add website title and stats controller

JohanneJayde committed:
added in remaining titles

JohanneJayde committed:
Replaced theme picker in sttings with ThemePicker component

JohanneJayde committed:
added watcher so that theme updates in all usages

JohanneJayde committed:
added in themed gradient background

JohanneJayde committed:
fix contrast in theme picker

JohanneJayde committed:
wordle buttons now react to theme change

alyxadarr committed:
Added "MasterOfTheUniverse" claim

alyxadarr committed:
Fixed Latest Wordles cutting off

alyxadarr committed:
Added a login page login.vue

alyxadarr committed:
Changed file name spelling, tied signindialog to app bar

alyxadarr committed:
Added policy for adding and removing words

alyxadarr committed:
Added claim to token controller

alyxadarr committed:
Added policy to program.cs

alyxadarr committed:
Changed favicon.ico

JohanneJayde committed:
styling Word Editor page

JohanneJayde committed:
adding EditWordDialog

JohanneJayde committed:
Updating EditWordDialog to render if if adding a word or editing one

JohanneJayde committed:
updaing common column in word editor

JohanneJayde committed:
restyling sign dialog

JohanneJayde committed:
adding in confirm didalog for deelting words and popuating editor dialog with word chosen

JohanneJayde committed:
add event emitters for when wordlist is edited

JohanneJayde committed:
delete extra console log statements

JohanneJayde committed:
Don't sort on Actions column and fix spelling typo

alyxadarr committed:
Added in Identity Seeder to program.cs

alyxadarr committed:
User b day migration and add to seeder

alyxadarr committed:
Birthday is added to JWT

JohanneJayde committed:
updating logo source change logic

JohanneJayde committed:
Using computed props for logo paths instead of watchres

JohanneJayde committed:
fix responsiveness in homepage and nav cards

JohanneJayde committed:
add textx wrap to page titles

JohanneJayde committed:
fix casing for standard logo svg

JohanneJayde committed:
more fixes to responsiveness of gird system in landing page

JohanneJayde committed:
UI Login Logout works

JohanneJayde committed:
general UI fixes

JohanneJayde committed:
fix issue with prerendering

JohanneJayde committed:
attempt to get login to show up again

alyxadarr committed:
Rename SigninDialog.vue to SignInDialog.vue

alyxadarr committed:
fixed spelling

JohanneJayde committed:
remove hw4 form other class file

JohanneJayde committed:
add WordEditorService with methods

JohanneJayde committed:
add routes to WordController not authorized yet

JohanneJayde committed:
register service

JohanneJayde committed:
add unique constarint for index on word

JohanneJayde committed:
fix edit word policy and authenticate CRUD actions

JohanneJayde committed:
remove ability to add words greater than 5 in AddWord

JohanneJayde committed:
endpoints are properly locked down

JohanneJayde committed:
WordList is now paginated on server side with search8ing

JohanneJayde committed:
update README.md to include claims for each user

JohanneJayde committed:
pagination and sarching workds on front end now

JohanneJayde committed:
fixed responsiveness of word editor

JohanneJayde committed:
add Edit andview mode into word editor

JohanneJayde committed:
only show appropriate actions for each account type

JohanneJayde committed:
added in updating common flag to front end

JohanneJayde committed:
add, update, and delete words work now

JohanneJayde committed:
fix typo

JohanneJayde committed:
updated GetWordsList to be ordered alphabetically

JohanneJayde committed:
use v-select to limit user from entering negative values for page size

JohanneJayde committed:
return totalCount to use actual pagination now

JohanneJayde committed:
Update README.md

Adde done requirements

JohanneJayde committed:
paingation is now on the botton of word editor now

JohanneJayde committed:
Merge branch 'Assignment5' of https://github.com/JohanneJayde/EWU-CSCD379-2024-Spring into Assignment5

JohanneJayde committed:
UI updsates

JohanneJayde committed:
updating table to be more responsive

JohanneJayde committed:
making login component responsive and clear inputs upon cancel

montalcy commented:
Ill review this :)

montalcy commented on Wordle.Api/Wordle.Api/Identity/Roles.cs:
```diff
@@ -2,4 +2,5 @@
 public static class Roles
 {
     public const string Admin = "Admin";
```
Just to clarify, awesome would technically be the one with the role MasterOfTheUniverse?

JohanneJayde replied:
Yes. I added it in a weird way so I'm going to update the way we add the claim so its easier to see.

montalcy commented on wordle-web/animations/fireworks.scss:
```diff
@@ -0,0 +1,124 @@
+@use "sass:math";
```
Really cool and creative styles!

JohanneJayde replied:
Thank you! @alyxadarr did that for Assignment3 😄

montalcy commented on wordle-web/components/SignInDialog.vue:
```diff
@@ -0,0 +1,86 @@
+          label="Password"
+          :type="showPassword ? 'text' : 'password'"
+          :append-inner-icon="
+            showPassword ? 'mdi-eye-off-outline' : 'mdi-eye-outline'
```
Nice addition we talked about in class

JohanneJayde replied:
Thank you! Yah it was cool when they did in class. It's pretty simple logic

montalcy reviewed (commented):
Create a page named "WordEditor" that allows words to be edited ✅
Support deleting a word ✅
Support adding a new word ✅
Editing words is not supported ✅
Duplicate words should not be allowed ✅ --- Would be cool if you added an error message so the user knows its a duplicate word :)
Support changing the common word boolean flag ✅
Words should be sorted alphabetically ✅
Provide a text box for searching words AS YOU TYPE which supports searching for words starting with letters specified ✅
Provide a menu item for navigating to the WordEditor page ✅
Anyone can look at the word list ✅
Word list is paginated (10-100 per page) ✅
Any logged in user can change the common word flag ✅
There is a login page/component/popup/etc for users to sign in ✅
Only users over 21 years of age (based on birthday) with a claim of MasterOfTheUniverse can add and remove words ✅   
Users can be set up at start up in the API project as a seed (apriori), there does not need to be a sign up mechanism ✅


editing bc I didnt see the last bullet -- y'all did great. I've loved seeing the progress on this wordle!!

connerhanson commented:
Ill review this as well!

connerhanson commented on Wordle.Api/Wordle.Api/Controllers/WordController.cs:
```diff
@@ -24,4 +27,38 @@ public async Task<string> GetWordOfDay(double offsetInHours = -7.0)
+    [HttpPost("EditWord")]
+    [Authorize]
+    public async Task EditWord(WordDto word)
+    {
```
Controllers all look great for the word Editor!

JohanneJayde replied:
Thank you 😄

connerhanson commented on Wordle.Api/Wordle.Api/Identity/IdentitySeed.cs:
```diff
@@ -43,5 +50,24 @@ private static async Task SeedAdminUserAsync(UserManager<AppUser> userManager)
         }
+
+        if (await userManager.FindByEmailAsync("Awesome@intellitect.com") == null)
+        {
```
This looks great, is a little funny that the admin account has less permissions than the other, but functionally wise it all looks great.

JohanneJayde replied:
Yah, I'm probably going to update this maybe if we decide to keep working on it past the end of this class.

connerhanson commented on Wordle.Api/Wordle.Api/Identity/Policies.cs:
```diff
@@ -0,0 +1,43 @@
+            if (birthdayString != null)
+            {
+                DateTime birthday;
+                if (DateTime.TryParse(birthdayString.Value, out birthday))
```
This works well, the only thing is that someone could be younger than 21 if their birthday hasn't happened yet that year.  You could fix this by doing something like I did in my attempt, Ill what I tried below: 

 int age = DateTime.Today.Year - paresedBDay.Year;
              // if this years date has not happened subtract a year from age
              if(DateTime.Today < paresedBDay.AddYears(age)){
                age--;
              }
              // else return if age is greater then 21
              return age >= 21;

JohanneJayde replied:
Yes I agree. I didn't even think about that until I did a review on Josh's and saw the checks he did. I will tri it out.

connerhanson commented on Wordle.Api/Wordle.Api/Services/WordEditorService.cs:
```diff
@@ -0,0 +1,55 @@
+
+    }
+
+    public async Task AddWord(WordDto wordToAdd)
```
Word editor services all look great!

JohanneJayde replied:
Thank you 
![a3f75e84b9542727230fc237ccc4e8d7](https://github.com/IntelliTect-Samples/EWU-CSCD379-2024-Spring/assets/114778753/37d70d0d-9801-44c3-a788-4e8e37d3c6ac)

connerhanson commented on wordle-web/components/EditWordDialog.vue:
```diff
@@ -0,0 +1,175 @@
+  }
+);
+
+const validLengthWod = (value: string) => {
```
The functionality seems to all work great from what I tested.

JohanneJayde replied:
![200w](https://github.com/IntelliTect-Samples/EWU-CSCD379-2024-Spring/assets/114778753/349159d8-af4f-4f76-8aec-ff104471d3d8)

Wahooooo

connerhanson reviewed (commented):
Overall this site looks great, its been exciting watching it evolve.  The only thing I could recommend fixing is for the website not to redirect you back to the home page/reset your theme when you login.  Otherwise everything looks great!

### Create an editor for the available words ✅
  - Create a page named "WordEditor" that allows words to be edited ✅
    - Support deleting a word ✅
    - Support adding a new word ✅
    - Editing words is not supported ✅
    - Duplicate words should not be allowed ✅
    - Support changing the common word boolean flag ✅ 
    - Words should be sorted alphabetically ✅
  - Provide a text box for searching words AS YOU TYPE which supports searching for words starting with letters specified ✅
  - Provide a menu item for navigating to the WordEditor page ✅
    - Anyone can look at the word list ✅
    - Word list is paginated (10-100 per page) ✅
  - Any logged in user can change the common word flag ✅
    - There is a login page/component/popup/etc for users to sign in ✅
  - Only users over 21 years of age (based on birthday) with a claim of MasterOfTheUniverse can add and remove words ✅
  - Users can be set up at start up in the API project as a seed (apriori), there does not need to be a sign up mechanism ✅

JohanneJayde committed:
addressed issue with not checking if someone's birthday is 21 if year difference is 0

Also updated Policy to be named AddOrDeleteWordPolicy as you can still edit a word's common flag even if you don't have this Policy

MikaellaCroskrey-IntelliTect commented:
## Assignment 5


### Create an editor for the available words ✅
  - Create a page named "WordEditor" that allows words to be edited ✅
    - Support deleting a word ✅
    - Support adding a new word ✅
    - Editing words is not supported ✅
    - Duplicate words should not be allowed ✅
    - Add & support changing the common word boolean flag ✅
      - Filter the list of returned words by common words ❌ Didn't see a way to filter the list in the UI or implemented code 
    - Words should be sorted alphabetically ✅
  - Provide a text box for searching words AS YOU TYPE which supports searching for words starting with letters specified ✅
  - Provide a menu item for navigating to the WordEditor page ✅
    - Anyone can look at the word list ✅
    - Word list is paginated on the server side (10-100 per page) ✅
  - Any logged in user can change the common word flag ✅
    - There is a login page/component/popup/etc for users to sign in ✅
  - Only users over 21 years of age (based on birthday) with a claim of MasterOfTheUniverse can add and remove words ✅ Great job on the claims!
  - Users can be set up at start up in the API project as a seed (apriori), there does not need to be a sign up mechanism ✅


## Extra Credit
Find a bug in the application, create an issue, submit a pull request against the issue ❌
Add the ability to sign up for an account by inputting an email and password ❌ Attempted but not finished?
Allow the user to elevate their level of privelage by inputting a secret phrase ❌
Select pagination record count ✅

Awesome work! The code looked really good, there was great attention to detail, and it had a very aesthetic UI. :)
